src/utils/temporal.hpp

```diff
@@ -414,13 +464,33 @@ struct ZonedDateTime {
   friend ZonedDateTime operator-(const ZonedDateTime &zdt, const Duration &dur) { return zdt + (-dur); }
 
   friend Duration operator-(const ZonedDateTime &lhs, const ZonedDateTime &rhs) {
-    return Duration(lhs.MicrosecondsSinceEpoch()) - Duration(rhs.MicrosecondsSinceEpoch());
+    return Duration(lhs.SysMicrosecondsSinceEpoch().count()) - Duration(rhs.SysMicrosecondsSinceEpoch().count());
+  }
+
+  int64_t LocalYear() const { return int(LocalYMD().year()); }
+  uint64_t LocalMonth() const { return static_cast<unsigned>(LocalYMD().month()); }
+  uint64_t LocalDay() const { return static_cast<unsigned>(LocalYMD().day()); }
```

Comment:
Can we use static_cast<> instead of c-style cast on line 470?
also can we just static cast into the numbered type? e.g. `...return static_cast<uint64_t>(...`?

Reply:
Fixed L470 to use `static_cast`. I use `static_cast<int>` and `static_cast<unsigned>` because other conversions aren’t available: `.year` returns `std::chrono::year`, which only has `operator int`. Analogously, `std::chrono::month` and `::day` only have `operator unsigned` available.
* https://en.cppreference.com/w/cpp/chrono/year
* https://en.cppreference.com/w/cpp/chrono/month
* https://en.cppreference.com/w/cpp/chrono/day